lib/bitcoin/lock/LockMonitor.ts

```diff
@@ -123,45 +117,45 @@ export default class LockMonitor {
   }
 
   private async handlePeriodicPolling (): Promise<void> {
+    this.currentLockState = await this.getCurrentLockState();
+    console.info(`Refreshed the in-memory value time lock state.`);
+
+    // If lock update is disabled, then no further action needs to be taken.
+    if (this.valueTimeLockUpdateEnabled === false) {
+      console.info(`Value time lock update is disabled, will not attempt to update the value time lock.`);
+      return;
+    }
 
     // If the current lock is in pending state then we cannot do anything and need to just return.
     if (this.currentLockState.status === LockStatus.Pending) {
       console.info(`The current lock status is in pending state; going to skip rest of the routine.`);
-
-      // But refresh the lock state before returning so that the next polling has the new value.
-      this.currentLockState = await this.getCurrentLockState();
       return;
     }
 
     // Now that we are not pending, check what do we have to do about the lock next.
     const validCurrentLockExist = this.currentLockState.status === LockStatus.Confirmed;
     const lockRequired = this.desiredLockAmountInSatoshis > 0;
 
-    let currentLockUpdated = false;
-
     if (lockRequired && !validCurrentLockExist) {
       await this.handleCreatingNewLock(this.desiredLockAmountInSatoshis);
-      currentLockUpdated = true;
     }
 
     if (lockRequired && validCurrentLockExist) {
       // The routine will true only if there were any changes made to the lock
-      currentLockUpdated =
-        await this.handleExistingLockRenewal(
-          this.currentLockState.activeValueTimeLock!,
-          this.currentLockState.latestSavedLockInfo!,
-          this.desiredLockAmountInSatoshis);
+      await this.handleExistingLockRenewal(
+        this.currentLockState.activeValueTimeLock!,
+        this.currentLockState.latestSavedLockInfo!,
+        this.desiredLockAmountInSatoshis
+      );
     }
 
     if (!lockRequired && validCurrentLockExist) {
-      currentLockUpdated =
-        await this.handleReleaseExistingLock(
-          this.currentLockState.activeValueTimeLock!,
-          this.desiredLockAmountInSatoshis);
-    }
+      console.info(LogColor.lightBlue(`Value time lock no longer needed.`));
 
-    if (currentLockUpdated) {

```

Comment:
Just wondering .. should we call the `getCurrentLockState` again at the end to make sure that we have the correct new state in case any updates to the lock were made? Any of the changes will not be reflected and the `this.currentLockState` will be in stale state for when the `getCurrentValueTimeLock` is called.

Off course if we remove the `this.currentLockState` completely then this issue goes away but that's a separate discussion above.

Reply:
Doing this twice in one method seems to be "bad design", which could be exactly why you have the initial fetch as the original code. This elevates the discussion of removing `this.currentLockState()` even more I think.